src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs

```diff
@@ -96,49 +105,81 @@ private void WriteParameter(CodeWriter writer, ServiceClientParameter clientPara
-        {
+        private string CreateNextPageRequestMethodName(ClientMethod operation) => $"Create{operation.Name}NextPageRequest";
+
+        private string GetNextPageMethodName(ClientMethod operation) => $"{operation.Name}NextPageAsync";
+
+        private bool IsNextPageParameter(ServiceClientParameter parameter) =>
+            parameter.Location != ParameterLocation.Path
+            && parameter.Location != ParameterLocation.Body
+            && parameter.Location != ParameterLocation.Query;
 
-            writer.Append($"internal {typeof(HttpMessage)} {CreateRequestMethodName(operation)}(");
-            foreach (ServiceClientParameter clientParameter in operation.Parameters)
+        private void WriteRequestCreation(CodeWriter writer, ClientMethod operation, bool nextPage = false)
+        {
+            var methodName = nextPage ? CreateNextPageRequestMethodName(operation) : CreateRequestMethodName(operation);
+            writer.Append($"internal {typeof(HttpMessage)} {methodName}(");
+            var parameters = nextPage
+                ? operation.Parameters.Where(IsNextPageParameter).ToArray()
+                : operation.Parameters;
+            foreach (ServiceClientParameter clientParameter in parameters)
             {
                 writer.Append($"{_typeFactory.CreateInputType(clientParameter.Type)} {clientParameter.Name},");
             }
+            if (nextPage)
+            {
+                writer.Append($"{writer.Type(typeof(string), true)} nextLinkUrl");
+            }
             writer.RemoveTrailingComma();
             writer.Line($")");
             using (writer.Scope())
             {
                 writer.Line($"var message = pipeline.CreateMessage();");
                 writer.Line($"var request = message.Request;");
                 var method = operation.Request.Method;
-                writer.Line($"request.Method = {writer.Type(typeof(RequestMethod))}.{method.ToRequestMethodName()};");
-
-                //TODO: Add logic to escape the strings when specified, using Uri.EscapeDataString(value);
-                //TODO: Need proper logic to convert the values to strings. Right now, everything is just using default ToString().
-                //TODO: Need logic to trim duplicate slashes (/) so when combined, you don't end  up with multiple // together
-                var urlText = String.Join(String.Empty, operation.Request.HostSegments.Select(s => s.IsConstant ? s.Constant.Value : "{" + s.Parameter.Name + "}"));
-                writer.Line($"request.Uri.Reset(new {writer.Type(typeof(Uri))}($\"{urlText}\"));");
+                writer.Line($"request.Method = {typeof(RequestMethod)}.{method.ToRequestMethodName()};");
 
-                foreach (var segment in operation.Request.PathSegments)
+                var uriType = writer.Type(typeof(Uri));
+                using (nextPage ? writer.If($"{uriType}.IsWellFormedUriString(nextLinkUrl, UriKind.Absolute)") : default)
```

Comment:
We don't need this logic, next link is always absolute when operation name is not set.

Reply:
Removed. Added https://github.com/Azure/autorest.csharp/issues/403